.github/workflows/plugin-check.yml

```diff
@@ -0,0 +1,53 @@
+name: 'Plugin Check'
+on:
+  pull_request:
+  push:
+    branches:
+    - main
+    - trunk
+    - 'release/**'
+
+
+jobs:
+  prepare:
+    name: "Build and check plugin"
+    # Don't run on forks.
+    if: github.repository_owner == 'Yoast'
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 7.2
+          coverage: none
+
+      - name: Set up node and enable caching of dependencies
+        uses: actions/setup-node@v3
+        with:
+          node-version: '14'
+          cache: 'yarn'
+
+      - name: Install Composer dependencies and run the prefixing script
+        uses: ramsey/composer-install@v2
+        with:
+          # Bust the cache at least once a week - output format: YYYY-MM-DD.
+          custom-cache-suffix: $(/bin/date -u --date='last Mon' "+%F")
+
+      - name: Yarn install
+        run: yarn install
+
+      - name: "Grunt: create artifact"
+        run: grunt artifact
+
+      - name: "Copy composer.json"
+        run: cp composer.json ./artifact
+
+      - name: Run plugin check
+        uses: swissspidy/wp-plugin-check-action@main
+        with:
+            build-dir: ./artifact
```

Comment:
One warning that's reported:

```
Warning: The TextDomain header in the plugin file does not match the slug. Found "wordpress-seo", expected "artifact".
```

We could do `mv artifact wordPress-seo` and then do `build-dir: './wordpress-seo'` to work around this.

Reply:
trying in  [9a8092a](https://github.com/Yoast/wordpress-seo/pull/20808/commits/9a8092aea4ff42633b0cd64622f930e94c206ed4)